snoop/data/models.py

```diff
@@ -138,21 +138,24 @@ def create_from_bytes(cls, data):
     @classmethod
     def create_from_file(cls, path):
         file_sha3_256 = hashlib.sha3_256()
-        with open(path, 'rb') as f:
-            for block in chunks(f):
-                file_sha3_256.update(block)
 
         try:
-            return Blob.objects.get(pk=file_sha3_256.hexdigest())
+            blob = Blob.objects.get(pk=file_sha3_256.hexdigest())
+            with open(path, 'rb') as f:
+                for block in chunks(f):
+                    file_sha3_256.update(block)
 
         except ObjectDoesNotExist:
             with cls.create() as writer:
                 with open(path, 'rb') as f:
                     for block in chunks(f):
+                        file_sha3_256.update(block)
```

Comment:
The hash is not used anywhere in this code. The `writer` will calculate its own hashes. How is this needed?